### ISSUE TITLE
empty workspaces do not need codelyzer rules

### ISSUE THREAD

**Issue by arjoreich** [angular/angular-cli#14130]
# 🐞 Bug report

### Command (mark with an `x`)
```
- [X] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Description
When creating an empty workspace, the recommended codelyzer rules should not be templated into the workspace's root tslint.json. 

### 🔬 Minimal Reproduction
1. `ng new repro-app --create-application=false`
2. `/repro-app/tslint.json` should not contain codelyzer specific rules.

### Desired Functionality
Move codelyzer rules to the tslint.json templates in the application & library schematics.

**Comment by alan-agius4** [comment]
Hi, can you kindly explain the reasoning behind this request? An Angular workspace will eventually contain a library or an application. 

Having them in the root `tslint.json` is the ideal way to ease maintenance and avoid having rules spread among various files.

**Comment by arjoreich** [comment]
> Hi, can you kindly explain the reasoning behind this request? An Angular workspace will eventually contain a library or an application.

 @alan-agius4 
I'm currently working on a reponse, I just didn't want the idea to be dismissed because I failed to properly make it's case.  I'll post it shortly...

**Comment by arjoreich** [comment]
_**TL;DR:** Its seems like a code smell that, after specifically choosing `ng-new --create-application=false` to move my app out of the root folder that the schematic should create application/library specific rules in the root `tslint.json` when there are no applications or libraries yet._

- _moves rules next to existing codelyzer  `directive-selector` & `component-selector` rules_
- _increases encapsulation and separations of concerns_
- _(potentially) makes it possible to have one sub-project stay on legacy rules & linter_
- _(potentially) while newly generated subj-project uses newer linter and recommendations_


> _"Having them in the root `tslint.json` is the ideal way to ease maintenance and avoid having rules spread among various files."_

While admittedly opinionated, I feel like the root `tslint.json` should be for `tslint:recommended` overrides and those settings the user deliberately decided need to be applied to all projects in his multi-project mono-repo.  I believe that it would be good code-stewardship to take on the burden as contributors of maintaining unit recommended rule lists for each applicable schematic and applying them to those templates to help guide users into the good programming habit of keeping their configurations as close to the code they cover.

Granted, in the default case where `createApplication=true` this is somewhat of a moot point because there exists and application on the same level as the root `tslint.json` However, I believe that it would be better code-stewardship if we encouraged users to configure their sub-projects as encapsulated as possible. 

In fact, the [Angular Guide -> Multiple project file structure](https://next.angular.io/guide/file-structure#multiple-project-file-structure) implies that using `createApplication=false` should limit the root to workspace related configurations.

**Comment by alan-agius4** [comment]
While I do agree that separation of concern is good and is the right approach. I personally don't feel that splitting the tslint rules fits in this, similarly to `tsconfig.json`. For these files we should use configuration inheritance to avoid unnecessary maintenance and also miss configurations. Since the more files you need to keep in sync, the greeter the room for error. 

The codelyzer settings that are defined in root `tslint.json` are the recommended settings for all the projects within a workspace.  I'd argue why a user would want different configurations per projects apart from the overrides that are needed for component and directive selector naming. I don't see it as a good practice to have different rules per projects since as all projects within a workspace one should have a consistent coding standard and hence share the same tslint rules.

I am going to put this up for discussion in our next team meeting.

**Comment by alan-agius4** [comment]
Hi, we just discussed this in our team meeting and we believe that using inherence for tslint.json is the right approach, as per explanation in the above comment.

Thanks.
